### PR TITLE
edit feature: suggest text in text fields

### PR DESCRIPTION
**Pull request by matallo**
can be tested in ded04, still wip tests pending

suggest

/cc @saleiva

### CODE REVIEW

**Comment by matallo** [comment]
tests are green (they were just fixed) but this includes no specific tests for the feature, will work on them

CR @xavijam to briefly check that utils for the column row fetch

File: lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-item.tpl [hunk 1]
```diff
@@ -8,5 +8,5 @@
     <span class="u-lSpace u-secondaryTextColor"><%- _t('components.backbone-forms.select.loading') %></span>
   </div>
 <% } else { %>
-  <%- label %>
+  <%- (label === null) ? 'null' : label %>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-item.tpl]: I'd not add this logic in the template, I'd pass the proper label text from the view.

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js [hunk 2]
```diff
@@ -25,14 +25,16 @@ module.exports = CoreView.extend({
     this.$el.empty();
     this.clearSubViews();
 
+    var name = this.model.getName() === null ? 'null' : this.model.getName();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js]: Could you check if this change will not break any other components? In my head I have the fill component, having items with the null value.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js]: I'm not sure, but just in case.

File: lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-view.js [hunk 3]
```diff
@@ -10,6 +10,10 @@ module.exports = CoreView.extend({
 
     this._generateForms();
 
+    this.model.bind('changeSchema', function () {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-view.js]:
```js
this.model.bind('changeSchema', this._generateForms, this);
```

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 var _ = require('underscore');
 var BaseAnalysisFormModel = require('./base-analysis-form-model');
 var template = require('./filter-form.tpl');
-var ColumnRowData = require('../column-row-data');
 var ColumnData = require('../column-data');
+var ColumnRowData = require('../../../../../data/column-row-data');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js]: Not needed change, I feel.

File: lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-view.js [hunk 5]
```diff
@@ -178,7 +178,9 @@ module.exports = CoreView.extend({
 
     this._attributesFormModel = new EditFeatureAttributesFormModel(this._featureModel.toJSON(), {
       featureModel: this._featureModel,
-      columnsCollection: self._sourceNode.querySchemaModel.columnsCollection
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-view.js]: Not sure, but do you need self in this context yet?

**Comment by xavijam** [comment]
Ah, tests are not included, when you have them, please let me know and I'll test and review the PR again.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js (hunk 4 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js]: uhm, needed, as I'm using this in other file (`edit-feature-attributes-form-model.js`), I just moved it from this folder to `/data`

File: lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-view.js (hunk 5 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-view.js]: self has been removed in this line

**Comment by matallo** [comment]
this is a first step, as explained, separated the Suggest editor from Select (thus, it won't affect current select components) but we are still relying in a re-render of the form to replace the text fields into Suggest fields, next step is doing this in the component itself (simlar to enabler editor?)

Anyway, I created the tests for this (struggled a lot 😞 ) and made some other small changes (when text in select item is only spaces, see attached)

screen shot 2016-11-22 at 18 30 36

to

screen shot 2016-11-24 at 17 31 16

CR @xavijam 🙏

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-add.tpl [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <div class="CDB-ListDecoration-item CustomList-item js-listItem">
   <button type="button" class="CDB-ListDecoration-itemLink CustomList-itemLink js-add-custom-value u-ellipsis" data-val="<%- query %>" title="<%- query %>">
     <p class="CDB-Text CDB-FontSize-small u-altTextColor"><%- _t('components.custom-list.add-custom-result') %></p>
-    “<%- query %>”
+    <%- query %>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-add.tpl]: Be careful with this edition, it will change the way/look of other items. cc @javierarce

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js (hunk 2 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js]: Same I said 3 days ago 💃

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-view.js [hunk 7]
```diff
@@ -139,7 +139,7 @@ module.exports = CoreView.extend({
 
     var mdl = this.collection.add({
       val: query,
-      label: '“' + query + '”'
+      label: query
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-view.js]: Same concern I've explained files above.

File: lib/assets/locale/en.json [hunk 8]
```diff
@@ -261,6 +261,11 @@
   },
   "form-components": {
     "editors": {
+      "radio": {
```
Comment (xavijam) [inline lib/assets/locale/en.json]: I feel there are labels already created with those parameters.

File: lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-model.js [hunk 9]
```diff
@@ -34,18 +39,48 @@ module.exports = Backbone.Model.extend({
...
       var columnName = mdl.get('name');
       var columnType = mdl.get('type');
 
       if (!_.contains(BLACKLISTED_COLUMNS, columnName)) {
+        // default field type
         this.schema[columnName] = {
           type: FIELDTYPE_TO_FORMTYPE[columnType]
         };
 
+        if (columnType === 'string') {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-model.js]: Do we still need this?

**Comment by xavijam** [comment]
Ah, ok, read it again your comment. Are we going to move that logic about String or Select to the component then?

File: lib/assets/locale/en.json (hunk 8 above)
Comment (matallo) [inline lib/assets/locale/en.json]: uh, I couldn't find them :/

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js (hunk 2 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-list-item-view.js]: I don't think so, tests are ok tho

summoning @javierarce, will check with him besides my own testing

File: lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-model.js (hunk 9 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-model.js]: we will need that line for sure, we would still need to move the text field to the suggest, but I'd do that in a separate PR

**Comment by matallo** [comment]
it can be tested in ded04, I just need to confirm with arce the null cases but I tested this extensively and it didn't break (plus, tests pass)

I reverted the use of apostrophes as they are currently used (see attached), will open the discusion in a separate issue

screen shot 2016-11-24 at 18 17 57

I tried to build the suggest editor as much as I could but having the text field in it would take me a lot longer, I prefer to do it in a separate PR

**Comment by matallo** [comment]
🚀
